api/discord.ts

```diff
@@ -0,0 +1,71 @@
+import { URLSearchParams } from 'url';
+
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { request } from 'undici';
+
+type AccessTokenResponse = {
+  access_token: string;
+  expires_in: number;
+  refresh_token: string;
+  scope: string;
+  token_type: 'Bearer';
+};
+
+type DiscordMember = {
+  id: string;
+  username: string;
+  discriminator: string;
+};
+
+export default async function handler(req: VercelRequest, res: VercelResponse) {
```

Comment:
Can you provide some docstring comments as a high-level overview of what this does? A link to the documentation where this general flow is detailed would probably suffice to provide most of the lower level detail (Some thing like [this](https://discord.com/developers/docs/topics/oauth2#authorization-code-grant))